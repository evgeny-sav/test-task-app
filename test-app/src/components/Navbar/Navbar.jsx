import React, { useEffect } from 'react';
import { NavLink, Link, useHistory } from 'react-router-dom';
import './Navbar.scss';
import { shallowEqual, useSelector, useDispatch } from 'react-redux';
import { logoutAction } from '../../store/actions/authActions';

const Navbar = () => {
  const dispatch = useDispatch();
  const history = useHistory();
  const { auth } = useSelector(state => {
    return {
      auth: {
        ...state.auth,
      },
    };
  }, shallowEqual);

  const handleLogout = e => {
    e.preventDefault();
    dispatch(logoutAction(auth.Token, history));
  };

  return (
    <div className="Navbar">
      <div className="brand">
        <Link to="/">Seagull AS</Link>
      </div>
      <div className="menulist">
        <ul>
          {auth.Token ? (
            <li>
              <NavLink to="/" activeClassName="active">
                Home
              </NavLink>
            </li>
          ) : null}
          {!auth.Token ? (
            <li>
              <NavLink to="/login" activeClassName="active">
                Login
              </NavLink>
            </li>
          ) : null}
          {auth.Token ? (
            <li>
              <NavLink to="/logout" onClick={handleLogout}>
                Logout
              </NavLink>
            </li>
          ) : null}
        </ul>
      </div>
    </div>
  );
};

export default Navbar;
