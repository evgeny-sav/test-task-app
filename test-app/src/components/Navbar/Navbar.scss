.Navbar {
  display: flex;
  flex-direction: row;
  height: 60px;
  background: #f8f9fa;
  margin-bottom: 10px;

  .brand {
    display: flex;
    width: 100px;
    justify-content: center;
    align-items: center;

    img {
      padding: 10px;
      max-width: 100%;
    }
  }

  .menulist {
    ul {
      margin: 0 0 0 10px;
      padding: 0;
      height: 100%;
      list-style-type: none;
      display: flex;
      flex-direction: row;

      li {
        display: flex;

        a {
          display: flex;
          justify-content: center;
          align-items: center;
          padding: 10px;
          text-decoration: none;
          color: #333333;

          &:hover {
            background: #e9ecef;
          }

          &.active {
            background: #e9ecef;
          }
        }
      }
    }
  }
}
