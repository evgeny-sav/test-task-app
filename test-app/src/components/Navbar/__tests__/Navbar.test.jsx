import React from 'react';
import { create } from 'react-test-renderer';
import Navbar from '../Navbar';

describe('Navbar', () => {
  it('should render', () => {
    const component = create(<Navbar />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
