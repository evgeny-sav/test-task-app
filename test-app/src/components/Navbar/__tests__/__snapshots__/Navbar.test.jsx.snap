// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render 1`] = `
<div
  className="Navbar"
>
  <div
    className="brand"
  >
    <Link
      to="/"
    >
      <img
        alt="Logo"
        src="pokemon_logo.png"
      />
    </Link>
  </div>
  <div
    className="menulist"
  >
    <ul>
      <li>
        <NavLink
          activeClassName="active"
          to="/news"
        >
          News
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="active"
          to="/contacts"
        >
          Contacts
        </NavLink>
      </li>
    </ul>
  </div>
</div>
`;
